relevanceai/clusterer/clusterer.py

```diff
@@ -961,3 +992,110 @@ def metadata(self, metadata: dict):
             alias=self.alias,
             metadata=metadata,
         )
+
+    def evaluate(
+        self,
+        ground_truth_column: Union[str, None] = None,
+        metric: Union[str, Callable] = "euclidean",
+        random_state: Union[int, None] = None,
+        average_score: bool = False,
+    ):
+        """
+        Evaluate your clusters using the silhouette score, or if true labels are provided, using completeness, randomness and homogeniety as well
+
+        Parameters
+        ----------
+        ground_truth_index : str
+            the index of the true label of each sample in dataset
+
+        metric : str or Callable
+            A string referencing suportted distance functions, or custom method for calculating the distance between 2 vectors
+
+        random_state : int, default None
+            for reproducability
+
+        average_score : bool
+            a boolean that determines whether to average the evaluation metrics in a new a score. only applicable if ground_truth_index is not None
+
+        Example
+        ----------
+
+        .. code-block::
+
+
+            from relevanceai import Client
+            client = Client()
+            df = client.Dataset("_github_repo_vectorai")
+            from relevanceai.clusterer import KMeansModel
+
+            model = KMeansModel()
+            kmeans = client.Clusterer(model, alias="kmeans_sample")
+            kmeans.fit(df, vector_fields=["sample_1_vector_"])
+
+            kmeans.evaluate()
+            kmeans.evaluate("truth_column")
+        """
+
+        vector_field = self.vector_fields[0]
+        cluster_field = self.cluster_field
+        alias = self.alias
+
+        samples = self._get_all_documents(self.dataset_id, include_vector=True)
+
+        vectors = [sample[vector_field] for sample in samples]
+        pred_labels = [sample[cluster_field][vector_field][alias] for sample in samples]
+
+        def norm(value, min, max):
+            return (value - min) / (max - min)
+
+        s_score = silhouette_score(
+            vectors, pred_labels, metric=metric, random_state=random_state
+        )
+
+        stats = {}
+        stats["silhouette"] = {
+            "score": s_score,
+            "description": METRIC_DESCRIPTION["silhouette"],
+        }
+
+        if ground_truth_column:
+            ground_truth_labels = {
```

Comment:
This code is a little hard to read - can you leave a comment on what ti's trying to do here?